### PR TITLE
Enable OpenStack Orchestration service (heat)

### LINKED ISSUE #61: Enable Heat service
Create recipe and use upstream cookbooks.

### PR DESCRIPTION
**Pull request by ramereth**
This should resolve #61.

*NOTE: For whatever reason the ServerSpec test for enabling the openstack-heat-api-cfn service fails. However if you converge again and run verify it passes. I can't figure out why this is happening but it might be a Chef bug itself.*

### CODE REVIEW

**Comment by ramereth** [comment]
[Enable OpenStack Orchestration service (heat)](https://trello.com/c/Gm4cKBtQ/567-enable-openstack-orchestration-service-heat)

**Comment by eldebrim** [comment]
It appears the orchestration-centos-74 suite doesn't converge 

           ================================================================================
           Error executing action `restart` on resource 'service[rabbitmq-server]'
           ================================================================================
           
           Mixlib::ShellOut::ShellCommandFailed
           ------------------------------------
           Expected process to exit with [0], but received '1'
           ---- Begin output of /bin/systemctl --system restart rabbitmq-server ----
           STDOUT: 
           STDERR: Job for rabbitmq-server.service failed because a fatal signal was delivered to the control process. See "systemctl status rabbitmq-server.service" and "journalctl -xe" for details.
           ---- End output of /bin/systemctl --system restart rabbitmq-server ----
           Ran /bin/systemctl --system restart rabbitmq-server returned 1
           
           Resource Declaration:
           ---------------------
           # In /tmp/kitchen/cache/cookbooks/rabbitmq/recipes/default.rb
           
           235:   service node['rabbitmq']['service_name'] do
           236:     action [:enable, :start]
           237:     supports :status => true, :restart => true
           238:     provider Chef::Provider::Service::Upstart if node['rabbitmq']['job_control'] == 'upstart'
           239:     provider Chef::Provider::Service::Init if node['rabbitmq']['job_control'] == 'init'
           240:   end
           241: else
           
           Compiled Resource:
           ------------------
           # Declared in /tmp/kitchen/cache/cookbooks/rabbitmq/recipes/default.rb:235:in `from_file'
           
           service("rabbitmq-server") do
             action [:enable, :start]
             updated true
             supports {:status=>true, :restart=>true}
             retries 0
             retry_delay 2
             default_guard_interpreter :default
             service_name "rabbitmq-server"
             enabled true
             pattern "rabbitmq-server"
             declared_type :service
             cookbook_name "rabbitmq"
             recipe_name "default"
           end
           
           Platform:
           ---------
           x86_64-linux
           
       
       Running handlers:
       [2017-10-09T16:24:21+00:00] ERROR: Running exception handlers
       Running handlers complete
       [2017-10-09T16:24:21+00:00] ERROR: Exception handlers complete
       Chef Client failed. 254 resources updated in 02 minutes 50 seconds
       [2017-10-09T16:24:21+00:00] FATAL: Stacktrace dumped to /tmp/kitchen/cache/chef-stacktrace.out
       [2017-10-09T16:24:21+00:00] FATAL: Please provide the contents of the stacktrace.out file if you file a bug report
       [2017-10-09T16:24:21+00:00] ERROR: Chef::Exceptions::MultipleFailures
       [2017-10-09T16:24:21+00:00] FATAL: Chef::Exceptions::ChildConvergeError: Chef run process exited unsuccessfully (exit code 1)

**Comment by ramereth** [comment]
@eldebrim what happens when you recreate the instance and try again? Also try removing ``Berksfile.lock`` to ensure you're getting the correct cookbooks.

**Comment by eldebrim** [comment]
@ramereth I removed `Berksfile.lock` and `~/.berkshelf/cookbooks` and recreated the instance, but I'm still getting the same issue. @osuosl-cookbooks/chefs is anyone else having the same issue, or could it just be my setup?

**Comment by ramereth** [comment]
@eldebrim are any of the other suites failing in a similar fashion? I.e. does the ops-messaging suite fail for you or the image suite?

**Comment by doublej472** [comment]
It looks like I can converge and verify on the second try, I was running into a different issue and ended up deleting my entire `~/.berkshelf` directory to get it working. That might help with the issue you are facing @eldebrim

File: test/integration/default/data_bags/x86_service_passwords/openstack-orchestration.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
+  "id": "openstack-orchestration",
+  "openstack-orchestration": {
+    "encrypted_data": "WLrdEYvl1Oj+vEhXu/BcTP2KkuI3I28Fu1i3zNwjlzld9rsajTB4WX9Xd/Xh\nzvkI\n",
+    "hmac": "GRMw/UWGn1rm5gIFMu5HEjLauQ0Wr4Zlezy2DAz4CNc=\n",
+    "iv": "GvU4RmsHJbjnEzrWFzMRow==\n",
+    "version": 2,
+    "cipher": "aes-256-cbc"
+  }
+}
```
Comment (doublej472) [inline test/integration/default/data_bags/x86_service_passwords/openstack-orchestration.json]: github complaining about a newline?

Comment (ramereth) [reply test/integration/default/data_bags/x86_service_passwords/openstack-orchestration.json]: I can't control this as it's created via knife. I would ignore this.

Comment (doublej472) [reply test/integration/default/data_bags/x86_service_passwords/openstack-orchestration.json]: How did the other data_bag get a newline? hmm...

**Comment by ramereth** [comment]
!bump minor openstack_newton

**Comment by osuosl-manatee** [comment]
Jenkins has merged this PR into `master` and has automatically performed a minor-level version bump to v3.1.0.  Have a nice day!

**Comment by eldebrim** [comment]
FWIW this (and any recipe that uses rabbitmq) failed for me because the hostname of my workstation (crimson) was too long for rabbitmq's config.

**Pull request merged**